agir/front/components/genericComponents/EventCard.js

```diff
@@ -225,7 +226,7 @@ const EventCard = (props) => {
   );
 
   return (
-    <StyledCard onClick={handleClick} $isPast={isPast}>
+    <StyledCard as={RenderIfVisible} onClick={handleClick} $isPast={isPast}>
```

Comment:
De ce que je comprends le `"as"  polymorphic prop` permet d'ajouter / transformer un héritage à la fin du rendu du composant.
Ce qui n'optimiserait pas le rendu de ce `StyledCard`, tu as pu tester ?

If you want to keep all the styling you've applied to a component but just switch out what's being ultimately rendered (be it a different HTML tag or a different custom component), you can use the "as" prop to do this at runtime.
https://styled-components.com/docs/api

Reply:
Ça a l'air de fonctionner comme ça, probablement parce que `Card` est un styled component aussi, et donc la valeur de `as` remplace le `styled.div`, plutôt que le `styled(Card)`